ports/esp32s2/common-hal/countio/Counter.c

```diff
@@ -0,0 +1,69 @@
+#include "common-hal/countio/Counter.h"
+
+#include "py/runtime.h"
+#include "supervisor/shared/translate.h"
+
+#include "driver/pcnt.h"
+
+static void pcnt_init(countio_counter_obj_t* self) {
+    int unit = PCNT_UNIT_0;
```

Comment:
Want to make this and rotaryio select units that aren't in use? That way we can multiple of each.

Reply:
Yes... I was already working on multiple unit implementation and I think I have got it figured out.